configure.ac

```diff
@@ -198,23 +198,57 @@ AC_ARG_WITH([cuse],
             [],
             [with_cuse=check]
 )
+AC_ARG_WITH([fuse3],
+            AS_HELP_STRING([--with-fuse3],[build with libfuse3 instead of libfuse2]),
+            [],
+            [with_fuse3=check]
+)
 
 if test "$with_cuse" != "no"; then
-    LIBFUSE_CFLAGS=$(pkg-config fuse --cflags 2>/dev/null)
-    if test $? -ne 0; then
-        if test "$with_cuse" = "yes"; then
-            AC_MSG_ERROR("Is fuse-devel installed? -- could not get cflags for libfuse")
-        else
-            with_cuse=no
-        fi
-    else
+    if test "$with_cuse" = "yes"; then
         with_cuse=yes
+    else
+        with_cuse=no
+    fi
+fi
+
+if test "$with_fuse3" != "no"; then
+    if test "$with_fuse3" = "yes"; then
+        with_fuse3=yes
+    else
+        with_fuse3=no
     fi
 fi
 
 dnl with_cuse is now yes or no
-if test "$with_cuse" != "no"; then
-    LIBFUSE_LIBS=$(pkg-config fuse --libs)
+dnl with_fuse3 is now yes or no
+if test "$with_cuse" == "yes" && test "$with_fuse3" == "yes"; then
+    LIBFUSE_CFLAGS="$(pkg-config fuse3 --cflags 2>/dev/null) -DWITH_FUSE3"
+    if test $? -ne 0; then
+        AC_MSG_ERROR("Is fuse3-devel installed? -- could not get cflags for libfuse3")
+    fi
+    LIBFUSE_LIBS=$(pkg-config fuse3 --libs 2>/dev/null)
+    if test $? -ne 0; then
+        AC_MSG_ERROR("Is fuse3-devel installed? -- could not get libs for libfuse3")
+    fi
+    AC_SUBST([LIBFUSE_CFLAGS])
+    AC_SUBST([LIBFUSE_LIBS])
+    AC_DEFINE_UNQUOTED([WITH_CUSE], 1,
+                       [whether to build with CUSE interface])
+
+    GTHREAD_LIBS=$(pkg-config --libs gthread-2.0)
```

Comment:
Why do you need gthread-2.0?  libfuse3 seems to need pthread:
```
$ pkg-config fuse3 --libs
-lfuse3 -lpthread
```

Reply:
You are right. If we compile with the CUSE interface then gthread is needed by the CUSE swtpm for a threadpool.